### PR TITLE
Fixes #2758: More accurate checks for dealer reg requests

### LINKED ISSUE #2758: Fix check_if_can_reg decorator to correctly check if post is from a dealer
In short, because we opened dealer registration before attendee registration, there was a loophole whereby a malicious attendee could preregister by posting a carefully crafted request. This pull request (https://github.com/magfest/ubersystem/pull/2756) was an effort to close that loophole, but it's an imperfect fix (see [this comment](https://github.com/magfest/ubersystem/pull/2756#discussion_r130252592) ).

Inside of `check_if_can_reg`, we should inspect the parameters for `PSEUDO_DEALER_BADGE`. If we do that then we can remove the additional check we do in `preregistration.form`.

### PR DESCRIPTION
**Pull request by RobRuana**

### CODE REVIEW

File: uber/templates/static_views/prereg_not_yet_open.html [hunk 1]
```diff
@@ -13,7 +13,7 @@ <h1 style='color:red'>{{ c.EVENT_NAME_AND_YEAR }} pre-registration is not yet op
   {%- if c.DEALER_REG_START and c.DEALER_REG_PUBLIC and c.BEFORE_DEALER_REG_SHUTDOWN -%}
   <p>
     <strong>Dealers:</strong>
-    {% if c.BEFORE_DEALER_REG_PUBLIC %}
+    {% if c.BEFORE_DEALER_REG_START %}
```
Comment (kitsuta) [inline uber/templates/static_views/prereg_not_yet_open.html]: Why was this changed? I thought the idea because dealer_reg_public that that it controls when this link appears.

Comment (RobRuana) [inline uber/templates/static_views/prereg_not_yet_open.html]: The intended use of these variables is not clear. `dealer_reg_public` defaults to the same value as `dealer_reg_start`. From what I can gather, `dealer_reg_public` will always either be the same as `dealer_reg_start` or an empty value to entirely turn off dealer registration visibility. If you notice, the outer conditional is checking the truthiness of `c.DEALER_REG_PUBLIC` to completely hide this block of HTML.

Let's assume for the moment that both `c.DEALER_REG_START` and `c.DEALER_REG_PUBLIC` are truthy values (as we expect they will be for most configurations, unless we are completely hiding dealer reg – in which case `c.DEALER_REG_PUBLIC` would be falsey, but we've already discussed how that is handled by the outer conditional). 

So both `c.DEALER_REG_START` and `c.DEALER_REG_PUBLIC` are truthy. There are three states we can have in this scenario:
* `c.DEALER_REG_START == c.DEALER_REG_PUBLIC` in which case, this distinction is moot
* `c.DEALER_REG_START < c.DEALER_REG_PUBLIC` dealer reg starts before it's public
* `c.DEALER_REG_START > c.DEALER_REG_PUBLIC` dealer reg is public before it starts

# `c.DEALER_REG_START < c.DEALER_REG_PUBLIC`

In this scenario, if a user loads the page in the time AFTER `c.DEALER_REG_START` but BEFORE `c.DEALER_REG_PUBLIC`, then they'll receive a message saying that dealer registration began some time in the past, but they'll have no way to navigate to the dealer application page.

This is obviously not what we want.

# `c.DEALER_REG_START > c.DEALER_REG_PUBLIC`

In this scenario, if a user loads the page in the time AFTER `c.DEALER_REG_PUBLIC` but BEFORE `c.DEALER_REG_START`, then they'll receive a link that says "Please register here". Upon clicking that link, they'll be redirected back to the same exact page because we haven't started accepting dealer registrations yet.

This is also obviously not what we want.

Thus, the correct value to check for that conditional is `c.BEFORE_DEALER_REG_START`.

Comment (kitsuta) [inline uber/templates/static_views/prereg_not_yet_open.html]: Huh, that's a good point.

I'm the one who added dealer_reg_public, and while MAGFest doesn't use it, the idea was that it would sometimes be set to after dealer_reg_start so that dealer reg could be 'soft launched' before we show the link. But as you mentioned, the text will display weirdly. The correct thing to do here then is either:

1. check for the truthiness of AFTER dealer_reg_public, not the truthiness of whether the date is set at all
2. display the reg start time as dealer_reg_public, not dealer_reg_start

That's a minor concern, though.

Comment (RobRuana) [inline uber/templates/static_views/prereg_not_yet_open.html]: Option 1. will work as long as dealer_reg_public is not set to an empty value. Option 2. is not a good idea, because we now have two variables controlling the same thing.
